bugfix: fix control plugin bug

luky116 opened the pull request:
# Pull Request Description

## What is the purpose of the change

当前 `applyTpsRule` 方法只检查 `rule` 是否为 null，但没有检查 `rule.getPointRule()` 是否为 null。当传入的 rule 对象不为 null 但其内部的 `pointRule` 为 null 时，会将无效的规则添加到规则集合中，可能导致后续处理异常或逻辑错误。

本次改动通过在 `applyTpsRule` 方法中增加对 `rule.getPointRule() == null` 的检查，确保只有当 rule 和 pointRule 都不为 null 时才添加规则，否则移除对应规则，提高代码的健壮性和安全性。

<img width="1481" height="553" alt="image" src="https://github.com/user-attachments/assets/0f58bfa1-a503-4f56-b750-a95057d7a1a8" />


## Brief changelog

- 修改 `DefaultTpsControlManager.applyTpsRule()` 方法，增加对 `rule.getPointRule() == null` 的检查
- 修改 `NacosTpsControlManager.applyTpsRule()` 方法，增加对 `rule.getPointRule() == null` 的检查

## Verifying this change

### 验证要点
1. **空规则处理验证**：当 rule 为 null 时，应该移除对应规则
2. **无效规则处理验证**：当 rule 不为 null 但 pointRule 为 null 时，也应该移除对应规则
3. **正常规则处理验证**：当 rule 和 pointRule 都不为 null 时，正常添加规则

### 代码质量检查
- ✅ Checkstyle 检查通过
- ✅ 代码逻辑正确，处理了边界情况

## Checklist

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.

* [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.

* [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.

* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
  - 建议添加单元测试验证 rule.getPointRule() 为 null 时的处理逻辑

* [x] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test` to make sure integration-test pass.
  - Checkstyle 检查通过
  - 代码编译通过

luky116 committed:
fix: 修复 TpsControlManager 相关问题

luky116 committed:
fix: 更新 NacosTpsControlManager

wuyfee commented:
$\color{red}{FAILURE}$
[DETAILS](https://github.com/alibaba/nacos/actions/runs/20496914302)
 ✅ - docker:   success 
 ❌ - deploy (standalone & cluster & standalone_auth):   failure 
 ❌ - e2e-java-test (standalone & cluster & standalone_auth):   skipped 
 ❌ - e2e-go-test (standalone & cluster):   skipped 
 ❌ - e2e-cpp-test (standalone & cluster):   skipped 
 ❌ - e2e-csharp-test (standalone & cluster):   skipped 
 ❌ - e2e-nodejs-test (standalone & cluster):   skipped 
 ❌ - e2e-python-test (standalone & cluster):   skipped 
 ✅ - clean (standalone & cluster & standalone_auth):   success

KomachiSion commented:
@luky116 CI unit test no pass, please fix.

luky116 committed:
fix: 修复 NacosTpsControlManagerTest.testApplyTpsRule1 测试失败

- 修改测试用例，为 TpsControlRule 设置 RuleDetail，使其符合当前实现逻辑
- 当前实现要求 rule.getPointRule() 不为 null 才会添加到 rules 中

luky116 committed:
Merge remote-tracking branch 'origin/develop' into fix-pluginBug-from-develop

luky116 committed:
fix

luky116 commented:
> @luky116 CI unit test no pass, please fix.

@KomachiSion  done

wuyfee commented:
$\color{red}{FAILURE}$
[DETAILS](https://github.com/alibaba/nacos/actions/runs/20567007835)
 ✅ - docker:   success 
 ❌ - deploy (standalone & cluster & standalone_auth):   failure 
 ❌ - e2e-java-test (standalone & cluster & standalone_auth):   skipped 
 ❌ - e2e-go-test (standalone & cluster):   skipped 
 ❌ - e2e-cpp-test (standalone & cluster):   skipped 
 ❌ - e2e-csharp-test (standalone & cluster):   skipped 
 ❌ - e2e-nodejs-test (standalone & cluster):   skipped 
 ❌ - e2e-python-test (standalone & cluster):   skipped 
 ✅ - clean (standalone & cluster & standalone_auth):   success
